tests/test_ilastik/test_applets/dataSelection/testOpDataSelection.py

```diff
@@ -658,198 +662,258 @@ def test_load_single_file_with_list(self):
         numpy.testing.assert_array_equal(imgData, self.imgData3Dct)
 
 
-def _make_data():
-    data_dict = {}
-    data_dict["rgb00c"] = numpy.array(
-        [
+class TestOpDataSelection_FileSeriesStacks:
+    @pytest.fixture(scope="class")
+    def tempdir(self, tmp_path_factory):
+        temp_dir = tmp_path_factory.mktemp("test_stack_along_data")
```

Comment:
Why does this work?.. New-style `tmp_path_factory` should return `pathlib.Path` instances that don't have `mktemp`.

Reply:
well, it's in [their example(https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-factory-fixture):

```Python
@pytest.fixture(scope="session")
def image_file(tmp_path_factory):
    img = compute_expensive_image()
    fn = tmp_path_factory.mktemp("data") / "img.png"
    img.save(fn)
    return fn
```

Reply:
Right, it's not just path itself but a path factory.

Reply:
`tmp_path` is the fixture that just gives you the Path